.ci/build-app-image.sh

```diff
@@ -12,7 +12,7 @@ BASE_IMAGE=$1
 TEST_APP=$2
 TAG=$3
 VERSION=$4
-
+VERSION=$(echo $VERSION | sed -e "s/-coreclr//")
```

Comment:
actually let's extract `$SAMPLES_REPO/samples/$VERSION/$TEST_APP` into a variable (it repeats at least twice) and handle it there inline like `".../${VERSION%-coreclr}/..."`